### PR TITLE
Feat: query simulate future reward

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

File: contracts/tokenomics/generator/src/contract.rs [hunk 1]
```diff
@@ -1326,6 +1341,111 @@ fn query_orphan_proxy_rewards(deps: Deps, lp_token: Addr) -> Result<Uint128, Con
...
+    lp_token: Addr,
+    future_block: u64,
+) -> Result<Uint128, ContractError> {
+    let cfg = CONFIG.load(deps.storage)?;
+
+    let lp_token = addr_validate_to_lower(deps.api, lp_token.as_str())?;
+    let pool = POOL_INFO.load(deps.storage, &lp_token)?;
+    let n_blocks = Uint128::from(future_block)
+        .checked_sub(env.block.height.into())
+        .unwrap_or_else(|_| Uint128::zero());
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:1438]: Imho, the user should know, that he entered incorrect `future_block` value. Provide informative custom exception.

File: contracts/tokenomics/generator/src/contract.rs [hunk 2]
```diff
@@ -1326,6 +1341,111 @@ fn query_orphan_proxy_rewards(deps: Deps, lp_token: Addr) -> Result<Uint128, Con
...
+        pending_astro_rewards = calculate_rewards(&env, &pool, &config)?;
+    }
+
+    // Calculate ASTRO tokens being distributed per block to this LP token pool
+    let astro_tokens_per_block: Uint128;
+    astro_tokens_per_block = config
+        .tokens_per_block
+        .checked_mul(Uint128::from(pool.alloc_point.u64()))?
+        .checked_div(Uint128::from(config.total_alloc_point.u64()))
+        .unwrap_or_else(|_| Uint128::zero());
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/tokenomics/generator/src/contract.rs:1401]: `unwrap_or_else` is better in this case. https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call

File: contracts/tokenomics/generator/src/contract.rs (hunk 1 above)
Comment (maksk7) [reply contracts/tokenomics/generator/src/contract.rs:1438]: > If the future block number passed to the function is actually in the past, return 0

File: contracts/tokenomics/generator/src/contract.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (wedancedalot) [reply contracts/tokenomics/generator/src/contract.rs:1401]: +1 for closure, that's what cargo clippy recommends

File: contracts/tokenomics/generator/src/contract.rs [hunk 3]
```diff
@@ -1326,6 +1341,111 @@ fn query_orphan_proxy_rewards(deps: Deps, lp_token: Addr) -> Result<Uint128, Con
...
+
+    let lp_token = addr_validate_to_lower(deps.api, lp_token.as_str())?;
+    let pool = POOL_INFO.load(deps.storage, &lp_token)?;
+
+    let lp_supply: Uint128;
+    let mut pending_on_proxy = None;
+    let mut pending_astro_rewards = Uint128::zero();
+
+    // If proxy rewards are live for this LP token, calculate its Pending Proxy rewards
+    match &pool.reward_proxy {
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/tokenomics/generator/src/contract.rs:1368]: Imho, there are no benefits to using `if let else` in this case.

**Pull request merged**